sdks/java/core/src/main/java/org/apache/beam/sdk/io/Compression.java

```diff
@@ -240,8 +244,66 @@ public WritableByteChannel writeCompressed(WritableByteChannel channel) throws I
       return Channels.newChannel(
           new SnappyCompressorOutputStream(Channels.newOutputStream(channel), uncompressedSize));
     }
+  },
+
+  /** Four MC Compression Splittable compressed file format leveraging LZ4
+   * Uses LZ4 Fast Compression*/
+  MC4_FAST(".4mc", ".4mc") {
+    @Override
+    public WritableByteChannel writeCompressed(WritableByteChannel channel) throws IOException {
+      return Channels.newChannel(
+              new FourMcCodec().createOutputStream(Channels.newOutputStream(channel)));
+    }
+
+    @Override
+    public ReadableByteChannel readDecompressed(ReadableByteChannel channel) {
+      return channel;
+    }
+  },
+  /** Four MC Compression Splittable compressed file format leveraging LZ4
+   * Uses LZ4 MC Compression*/
+  MC4_MEDIUM(".4mc", ".4mc") {
```

Comment:
Please add tests for these compression formats. See CompressedSourceTest for similar tests.